packages/system/src/system.ts

```diff
@@ -40,16 +46,31 @@ interface GetStyleObject {
  */
 export const toCSSObject: GetStyleObject = ({ baseStyle }) => (props) => {
   const { theme, css: cssProp, __css, sx, ...rest } = props
-  const styleProps = objectFilter(rest, (_, prop) => isStyleProp(prop))
-  const finalBaseStyle = runIfFn(baseStyle, props)
-  const finalStyles = Object.assign(
-    {},
-    __css,
-    finalBaseStyle,
-    filterUndefined(styleProps),
-    sx,
+  const stylePrecedence = []
+
+  // Styles are applied by precedence (lowest to highest)
+  //   __css
+  //   baseStyle
+  //   other (apply, layerStyles, textStyles)
+  //   styleProps
+  //   sx
+  //   css
+  stylePrecedence.push({ ...__css })
```

Comment:
Could we do without spreading the object and just push to the `stylePrecedence` array directly?